### ISSUE TITLE
ApplicationSet git generator should expose {{filePath}} param, not just folder based params.

### ISSUE THREAD

**Issue by yalimpop** [argoproj/argo-cd#9465]
# Summary

Add `{{filepath}}` to available params when git files generator is templated.

# Motivation
We use ApplicationSet with git generator to discover multiple helm values files(`*.yaml`), and run a custom helm plugin for each file found. The helm plugin that we wrote is quite simple, and basically uses `helm template -f {filePath}` to render the Kubernetes resource files. 
This is a different concept than I believe was intended by the git files generator, but it works amazingly into our use case. 

# Proposal

A short addition to the params exposed at the git generator.

We are already working with a custom image build from [this PR](https://github.com/argoproj/argo-cd/pull/9463).

I am working on passing and adding the required tests set by the [contributing guidelines](https://argo-cd.readthedocs.io/en/latest/developer-guide/code-contributions/)

**Comment by iamnoah** [comment]
I think  #9839 addresses this

**Comment by crenshaw-dev** [comment]
Agreed, I think you can get what you need with `{{path}}/{{path.filename}}` after @iamnoah's fix in 2.4.4.

I do see the benefit of a dedicated `filepath` param though, if you still want to add that.
